### PR TITLE
Rweber/opencl

### PR DESCRIPTION
**Pull request by rickwebiii**
Correctly computes bins for each window.

### CODE REVIEW

File: sunscreen_math/src/opencl_impl/ristrettovec.rs [hunk 1]
```diff
@@ -76,6 +79,16 @@ impl GpuRistrettoPointVec {
         }
     }
 
+    /// Allocate a new [`RistrettoPointVev`] initialized to zero.
```
Comment (samtay) [inline sunscreen_math/src/opencl_impl/ristrettovec.rs:82]: typo `Vev` -> `Vec`

File: sunscreen_math/src/opencl_impl/multiexp.rs [hunk 2]
```diff
@@ -368,4 +403,65 @@ mod tests {
...
+        let actual_gpu = compute_bucket_points(&points_gpu, &bucket_data, window_size_bits);
+
+        let actual = actual_gpu.iter().collect::<Vec<_>>();
+
+        let expected = crate::test_impl::compute_bucket_points(&scalars, &points, window_size_bits);
+
+        let num_buckets = multiexp_num_buckets(window_size_bits);
+
+        for (actual_window_buckets, expected_window_buckets) in
+            actual.chunks(num_buckets).zip(expected)
```
Comment (samtay) [inline sunscreen_math/src/opencl_impl/multiexp.rs:460]: This does skip testing that `actual.len() / num_buckets == expected.len()`, unsure if that's important.

Comment (rickwebiii) [reply sunscreen_math/src/opencl_impl/multiexp.rs:460]: Better safe than sorry.

File: sunscreen_math/src/opencl_impl/mod.rs [hunk 3]
```diff
@@ -284,7 +284,6 @@ impl Runtime {
 
     fn alloc_internal<T: OclPrm>(&self, len: usize) -> Buffer<T> {
         Buffer::builder()
-            .fill_val(T::default()) // Avoids dealing with MaybeUninit<T>
```
Comment (rickwebiii) [inline sunscreen_math/src/opencl_impl/mod.rs:287]: Oh, now I need to propagate this function as unsafe...

**Pull request merged**